lib/index.js

```diff
@@ -230,4 +214,27 @@ module.exports = {
     'promise/catch-or-return': 'error',
     'promise/no-new-statics': 'error',
   },
+},
+{
+  files: ['test/**'],
+  rules: {
+    'no-global-assign': 'off',
+    'no-extend-native': 'off',
+  },
+}, {
+  files: ['lib/**', 'bin/**'],
+  rules: {
+    'import/no-extraneous-dependencies': ['error', { devDependencies: false }],
+  },
+}, {
+  files: ['lib/**'],
+  rules: {
+    'no-console': 'error',
+  },
```

Comment:
Add descriptive names to these to more easily identify them in eg. the config inspector, eg:

```suggestion
{
  name: 'Test adaptions',
  files: ['test/**'],
  rules: {
    'no-global-assign': 'off',
    'no-extend-native': 'off',
  },
}, {
  name: 'Shipping code adaptions',
  files: ['lib/**', 'bin/**'],
  rules: {
    'import/no-extraneous-dependencies': ['error', { devDependencies: false }],
  },
}, {
  name: 'Shipping code adaptions, non-cli',
  files: ['lib/**'],
  rules: {
    'no-console': 'error',
  },
```